internal/invocation/invocation_service.go

```diff
@@ -78,6 +83,20 @@ func NewService(handler Handler, ed *event.DispatchService, lg logger.LogAdaptor
 			}
 		}()
 	})
+	s.eventDispatcher.Subscribe(EventInvocationStateChanged, stateSubID, func(event event.Event) {
+		e := event.(*InvocationStateChanged)
```

Comment:
Those events won't get lost. The "normal" invocations are not accepted to the invocation channel if the invocation service is paused: https://github.com/yuce/hazelcast-go-client/blob/ad4231ffbac8ec70c57a8c235e732b44b415cb0c/internal/invocation/invocation_service.go#L120

Reply:
OK, not losing them is good. But, the second part of my question remains open. In the time that passed between we call the publish for this event and it starts executing, we allow invocations, right? If so, I believe that is a problem

Reply:
Invocation service starts paused, and only enabled after the client connects to the client, so during that time there's no chance of sending non-urgent invocations.

Invocation service is paused when the client is disconnected from the cluster (all connections are removed). If an invocation is accepted after all connections are removed and before `InvocationStateChanged` event is received, the invocation wouldn't be send to any connection, since there wouldn't be any live connections to the cluster at that stage.

Reply:
Event order is guaranteed: https://github.com/hazelcast/hazelcast-go-client/blob/180f5ebe9c878e0ec1aed02a2edf63339e929668/internal/event/dispatch_service_test.go#L99

Reply:
OK, that's good. But, we still have a problem with the unknown scheduling of the event callbacks

Reply:
You are right. Refactored that at: https://github.com/hazelcast/hazelcast-go-client/pull/910/commits/154aa2354f6296bc05f16d6c7df8fa1d4c725002